From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: HaHaWTH <102713261+HaHaWTH@users.noreply.github.com>
Date: Tue, 9 Nov 2077 00:00:00 +0800
Subject: [PATCH] propagate block destruction cancellation

Paper issue: https://github.com/PaperMC/Paper/issues/13586

License: GPL-3.0-only (https://www.gnu.org/licenses/gpl-3.0.html)

Additional term under GNU GPLv3 section 7(b):

Any copy or modified version containing a substantial portion of the
original contributions in this patch must preserve the following
author attribution in the patch header, source code, or another
prominent legal notice accompanying the corresponding source:

Original patch by HaHaWTH <102713261+HaHaWTH@users.noreply.github.com>

diff --git a/src/main/java/org/bukkit/craftbukkit/util/DelegatedLevelAccessor.java b/src/main/java/org/bukkit/craftbukkit/util/DelegatedLevelAccessor.java
index 1dbbfa51226c69e61564d40a804a82c184d6f3ef..2a99d92db7b7b9f9e5daed8e6c242e8561665fbd 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/DelegatedLevelAccessor.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/DelegatedLevelAccessor.java
@@ -674,6 +674,13 @@ public abstract class DelegatedLevelAccessor implements WorldGenLevel {
         return this.delegate.destroyBlock(pos, dropResources, breaker, updateLimit);
     }
 
+    // Leaf start - propagate block destruction cancellation
+    @Override
+    public net.minecraft.world.level.LevelWriter.DestroyBlockResult destroyBlockWithResult(BlockPos pos, boolean dropResources, @Nullable Entity breaker, int updateLimit) {
+        return this.delegate.destroyBlockWithResult(pos, dropResources, breaker, updateLimit);
+    }
+    // Leaf end - propagate block destruction cancellation
+
     @Override
     public boolean addFreshEntity(Entity entity) {
         return this.delegate.addFreshEntity(entity);
